lightning/src/ln/channelmanager.rs

```diff
@@ -83,8 +97,17 @@ enum PendingHTLCRouting {
 
 #[derive(Clone)] // See Channel::revoke_and_ack for why, tl;dr: Rust bug
 pub(super) struct PendingHTLCInfo {
+	#[cfg(test)]
+	pub(super) routing: PendingHTLCRouting,
+	#[cfg(not(test))]
 	routing: PendingHTLCRouting,
+	#[cfg(test)]
+	pub(super) incoming_shared_secret: [u8; 32],
+	#[cfg(not(test))]
 	incoming_shared_secret: [u8; 32],
+	#[cfg(test)]
+	pub(super) payment_hash: PaymentHash,
+	#[cfg(not(test))]
```

Comment:
Do these conditionally need to be `pub(super)`? I see `amt_to_forward` and `outgoing_cltv_value` are not, so I'm wondering if those are like that for testing.

Reply:
got rid of this, thanks